src/main/cli/rpc.ts

```diff
@@ -0,0 +1,374 @@
+      console.timeEnd(`makeRpcRequest ${method}`);
+    }
+  });
+}
+
+export async function makeBatchRpcRequest<T>(
+  method: RpcMethod,
+  params: jayson.RequestParamsLike[],
+  logCallback?: (log: CliLog[]) => void
+): Promise<T[]> {
+  return new Promise<T[]>(async (resolve, reject) => {
+    console.time(`makeRpcRequest ${method} (batch)`);
+    store.dispatch(rpcSetEndpointBusy(method));
+
+    if (logCallback) {
+      const logReferenceId = uuidv4();
+
+      params.forEach((param) => {
+        if (isObject(param)) {
+          Object.assign(param, { log_reference_id: logReferenceId });
+        }
+      });
+
+      RPC_LOG_EVENT_EMITTER.on((logs) => {
+        const relevantLogs = logs.filter(
+          (log) => getCliLogSpanLogReferenceId(log) === logReferenceId
+        );
+        logCallback(relevantLogs);
+      });
+    }
+
+    const batch = params.map((param) =>
+      rpcClient.request(method, param, undefined, false)
+    );
+
+    try {
+      const responses = (await rpcClient.request(batch)) as RawRpcResponse<T>[];
+
+      // Check if any of the responses have an error
+      const errorResponses = responses.filter(isErrorResponse);
+      if (errorResponses.length > 0) {
+        throw new Error(
+          `One or more RPC requests failed Errors: ${errorResponses
+            .map((r) => `${r.error.message} (${r.error.code})`)
+            .join(', ')}`
+        );
+      }
+
+      logger.debug(
+        { method, length: responses.length },
+        'Received batched RPC response (success)'
+      );
+
+      // Because we know that all responses are successful, we can cast them to RawRpcResponseSuccess
+      const results = (responses as RawRpcResponseSuccess<T>[]).map(
+        (r) => r.result
+      );
+      resolve(results);
+    } catch (e) {
+      reject(e);
+    } finally {
+      store.dispatch(rpcSetEndpointFree(method));
+      console.timeEnd(`makeRpcRequest ${method} (batch)`);
+    }
+  });
+}
+
+function providerFromGetSellerResponse(
+  providerResponse: SwapSellerInfo
+): Provider {
+  const multiAddr = new Multiaddr(providerResponse.addresses[0])
+    .decapsulate('p2p')
+    .toString();
+
+  return {
+    peerId: providerResponse.peerId,
+    multiAddr,
+    testnet: isTestnet(),
+  };
+}
+
+export async function checkBitcoinBalance() {
+  const response = await makeRpcRequest<BalanceBitcoinResponse>(
+    RpcMethod.GET_BTC_BALANCE,
+    {}
+  );
+  store.dispatch(rpcSetBalance(response.balance));
+}
+
+export async function withdrawAllBitcoin(address: string) {
+  store.dispatch(rpcResetWithdrawTxId());
+  const response = await makeRpcRequest<WithdrawBitcoinResponse>(
+    RpcMethod.WITHDRAW_BTC,
+    {
+      address,
+    }
+  );
+  store.dispatch(rpcSetWithdrawTxId(response.txid));
+}
+
+export async function getSwapInfo(swapId: string) {
+  return makeRpcRequest<GetSwapInfoResponse>(RpcMethod.GET_SWAP_INFO, {
+    swap_id: swapId,
+  });
+}
```

Comment:
**Suggestion:** Consider adding error handling for the async function 'getSwapInfo' at line 198. Currently, if an error occurs during the execution of 'makeRpcRequest', it will not be caught and could lead to unexpected behavior.
```suggestion
export async function getSwapInfo(swapId: string) {
  try {
    return makeRpcRequest<GetSwapInfoResponse>(RpcMethod.GET_SWAP_INFO, {swap_id: swapId});
  } catch (e) {
    logger.error(`Error occurred: ${e.message}`);
  }
}
```